odiglet/pkg/kube/runtime_details/shared.go

```diff
@@ -65,32 +64,37 @@ func runtimeInspection(pods []corev1.Pod) ([]common.LanguageByContainer, error)
 				continue
 			}
 
-			detectionResults := inspectors.DetectLanguage(processes)
-			if len(detectionResults) == 0 {
-				log.Logger.V(0).Info("no supported language detected for container in pod", "processes", processes, "pod", pod.Name, "container", container.Name, "namespace", pod.Namespace)
-				continue
-			}
-
-			if len(detectionResults) > 1 {
-				log.Logger.V(0).Info("multiple languages detected for pod container processes, selecting first one", "processes", processes, "pod", pod.Name, "container", container.Name, "namespace", pod.Namespace)
-			}
-
-			resultsMap[container.Name] = common.LanguageByContainer{
-				ContainerName: container.Name,
-				Language:      detectionResults[0].Language,
+			for _, process := range processes {
+				lang, err := inspectors.DetectLanguage(process)
+				if err != nil {
+					log.Logger.V(0).Info("no supported language detected for container in pod", "process", process, "pod", pod.Name, "container", container.Name, "namespace", pod.Namespace)
+					continue
+				}
+
+				// Convert map to slice for k8s format
+				envs := make([]odigosv1.EnvVar, 0, len(process.Envs))
+				for envName, envValue := range process.Envs {
+					envs = append(envs, odigosv1.EnvVar{Name: envName, Value: envValue})
+				}
+
+				resultsMap[container.Name] = odigosv1.RuntimeDetailsByContainer{
+					ContainerName: container.Name,
+					Language:      lang,
+					EnvVars:       envs,
+				}
```

Comment:
Since we don't have CRD supporting multiple processes inside the same container, we don't have a good way of handling it currently. I think the the behavior before this PR is to take into account the first process and here it is the last process, both options are not ideal. Do you think we should consider the first process only?